### PR TITLE
[4.0]namespace administrator components com_languages

### PR DESCRIPTION
**Pull request by brianteeman**
continuing the work to namespace joomla4 one admin component at a time.

please wait for drone to successfully complete before reviewing.

### CODE REVIEW

File: administrator/components/com_languages/tmpl/override/edit.php [hunk 1]
```diff
@@ -9,29 +9,32 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\HTML\HTMLHelper;
```
Comment (Quy) [inline administrator/components/com_languages/tmpl/override/edit.php]: Remove duplicate

Comment (brianteeman) [reply administrator/components/com_languages/tmpl/override/edit.php]: done

File: administrator/components/com_languages/Model/StringsModel.php [hunk 2]
```diff
@@ -120,7 +123,7 @@ public function refresh()
 	public function search()
 	{
 		$results = array();
-		$input   = \JFactory::getApplication()->input;
+		$input   = Factory::getApplication()->input;
 		$filter  = \JFilterInput::getInstance();
```
Comment (Quy) [inline administrator/components/com_languages/Model/StringsModel.php]: Update with use

File: administrator/components/com_languages/tmpl/override/edit.php [hunk 3]
```diff
@@ -10,28 +10,30 @@
...
-JHtml::addIncludePath(JPATH_COMPONENT . '/helpers/html');
+HTMLHelper::addIncludePath(JPATH_COMPONENT . '/helpers/html');
 
-JHtml::_('behavior.formvalidator');
-JHtml::_('behavior.keepalive');
+HTMLHelper::_('behavior.formvalidator');
+HTMLHelper::_('behavior.keepalive');
 
 
 $expired = ($this->state->get('cache_expired') == 1 ) ? '1' : '';
```
Comment (Quy) [inline administrator/components/com_languages/tmpl/override/edit.php:22]: Move after all `HTMLHelper`s if you want.

File: administrator/components/com_languages/View/Overrides/HtmlView.php [hunk 4]
```diff
@@ -101,37 +105,37 @@ protected function addToolbar()
...
-			\JToolbarHelper::preferences('com_languages');
+			ToolbarHelper::preferences('com_languages');
 		}
 
-		\JToolbarHelper::divider();
-		\JToolbarHelper::help('JHELP_EXTENSIONS_LANGUAGE_MANAGER_OVERRIDES');
+		ToolbarHelper::divider();
+		ToolbarHelper::help('JHELP_EXTENSIONS_LANGUAGE_MANAGER_OVERRIDES');
 
 		\JHtmlSidebar::setAction('index.php?option=com_languages&view=overrides');
```
Comment (Quy) [inline administrator/components/com_languages/View/Overrides/HtmlView.php:133]: Use `HTMLHelper`

File: administrator/components/com_languages/View/Overrides/HtmlView.php [hunk 5]
```diff
@@ -101,37 +105,37 @@ protected function addToolbar()
...
 
 		\JHtmlSidebar::addFilter(
```
Comment (Quy) [inline administrator/components/com_languages/View/Overrides/HtmlView.php:135]: Use `HTMLHelper`

Comment (brianteeman) [reply administrator/components/com_languages/View/Overrides/HtmlView.php:135]: I deliberately ignored all references to the sidebar as they are scheduled to be removed

File: administrator/components/com_languages/View/Overrides/HtmlView.php (hunk 4 above)
Comment (brianteeman) [reply administrator/components/com_languages/View/Overrides/HtmlView.php:133]: I deliberately ignored all references to the sidebar as they are scheduled to be removed

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7b2ad2929dc403384053192e206816e40abd1b95<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21058">issues.joomla.org/tracker/joomla-cms/21058</a>.</sub>

File: administrator/components/com_languages/Model/OverridesModel.php [hunk 6]
```diff
@@ -15,6 +15,9 @@
 use Joomla\CMS\MVC\Model\ListModel;
 use Joomla\CMS\MVC\Factory\MVCFactoryInterface;
 use Joomla\Component\Languages\Administrator\Helper\LanguagesHelper;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Fileystem\File;
```
Comment (Quy) [inline administrator/components/com_languages/Model/OverridesModel.php]: Spelling `Fileystem`

**Comment by brianteeman** [comment]
conflict resolved

**Comment by brianteeman** [comment]
conflict resolved

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d6c5e5a54fee824375e0f66ea8a6e4e0feb268e7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21058">issues.joomla.org/tracker/joomla-cms/21058</a>.</sub>

**Comment by brianteeman** [comment]
@wilsonge can this one be merged yet?

**Comment by wilsonge** [comment]
Yup - sorry must have missed it - thanks!

**Comment by brianteeman** [comment]
thanks

**Pull request merged**